### ISSUE TITLE
Undefined: block timestamp validation logic

### ISSUE THREAD

**Issue by whyrusleeping** [filecoin-project/specs#458]
Its not clear how to validate a timestamp on a block header. The spec says it must be at least one block time past the previous block, and 'not in the future'.

I think those rules are likely incorrect, as clock skew would likely cause miners to drop a lot of blocks.

**Comment by anorth** [comment]
There's a `ALLOWABLE_CLOCK_DRIFT` in https://github.com/filecoin-project/specs/blob/master/mining.md#block-validation, but IMO the logic there should be expressed only once, in prose, in https://github.com/filecoin-project/specs/blob/master/validation.md (which I think is where you were looking)

**Comment by sternhenri** [comment]
Yes. Consolidation is right. Good catch (I’d never seen validation.md :b)

**Comment by ZenGround0** [comment]
@whyrusleeping should be closing this out in #697, will post a link once merged.
